### PR TITLE
Add CI tests for web and contracts

### LINKED ISSUE #11: automated testing thru Github Actions
we already have CI for deployment thru Cloudflare Pages

now we need CI for unit testing

```
cd packages/portal-contracts && forge test --vv
cd packages/portal-web && npm ci && npm test
```

### PR DESCRIPTION
**Pull request by kahuang**
Adds github actions for running the web/contracts tests. Fixes #11

### CODE REVIEW

File: .github/workflows/contracts.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./packages/portal-contracts
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
```
Comment (dcposch) [inline .github/workflows/contracts.yml:23]: cool that they have a prebuilt nightly image

File: .github/workflows/web.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm test
```
Comment (dcposch) [inline .github/workflows/web.yml:29]: 👍

File: .github/workflows/web.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (dcposch) [inline .github/workflows/web.yml:27]: oh, fancy. tbh testing with latest node is probably fine

File: packages/portal-contracts/.gas-snapshot [hunk 4]
```diff
@@ -0,0 +1,8 @@
+PortalTest:testAsk():(address) (gas: 1693305)
+PortalTest:testBid():(address) (gas: 1693434)
+PortalTest:testBuy():(address) (gas: 1762472)
+PortalTest:testCancel() (gas: 1652417)
+PortalTest:testSell():(address) (gas: 1776687)
+PortalTest:testSetParam() (gas: 1595330)
+PortalTest:testSettle() (gas: 1935749)
+PortalTest:testSlash() (gas: 1698649)
```
Comment (dcposch) [inline packages/portal-contracts/.gas-snapshot:8]: foundry is so great

**Review by dcposch — APPROVED** [review]
Looks excellent, thank you!

**Comment by kahuang** [comment]
I got a little confused with the PR UI and ended up merging this here: https://github.com/dcposch/silverportal/pull/14

Closing this one

**Pull request closed without merging**